lib/core/task.toit

```diff
@@ -75,6 +76,90 @@ interface Task:
   static current -> Task:
     return Task_.current
 
+  /**
+  Divides the work given by $lambdas up into a number of
+    separate tasks that run concurrently.
+
+  Returns the results of running the $lambdas as a $Map
+    keyed by the lambda index. If the lambda at a given
+    index did not return, the results map will not contain
+    and entry for it. The results map is insertion ordered,
+    so it is possible to tell which lambda returned first.
+
+  If any of $lambdas throw an exception, the exception is
+    propagated to the caller of $Task.divide which in
+    returns also throws.
+
+  If $required is less than the number of $lambdas, the
+    method returns when $required tasks have completed.
+  */
+  static divide lambdas/List -> Map
```

Comment:
?
```suggestion
  static run_in_parallel lambdas/List -> Map
```

Reply:
I thought about such names, but it does much more than just run *concurrently*. It is the basis for a lot of divide and conquer operations, so I thought about things like split, join, or merge, but I couldn't think of something much better than raw `divide` to indicate that you're dividing one task up into many dependent ones.